### PR TITLE
Set the window icon with sdl2

### PR DESCRIPTION
**Pull request by onon1101**
這主要新增可以設定 window 的 icon，以下是範例。

截圖 2024-04-09 下午3 30 22

### CODE REVIEW

**Review by NOOBDY — CHANGES_REQUESTED** [review]

**Comment by onon1101** [comment]
Do i need to push again ?

**Comment by onon1101** [comment]
i don't know why it's not pass in macOS

**Comment by NOOBDY** [comment]
i have triggered a re-run for macos workflow

File: .gitignore [hunk 1]
```diff
@@ -6,6 +6,10 @@
 # Covers JetBrains IDEs: IntelliJ, RubyMine, PhpStorm, AppCode, PyCharm, CLion, Android Studio, WebStorm and Rider
 # Reference: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839
 
+# Ignore .DS_Store file from a git repository
+
+./**/.DS_Store
```
Comment (ntut-rick) [inline .gitignore:11]:
```suggestion
.DS_Store
```

**Comment by onon1101** [comment]
MR

**Review by ntut-rick — APPROVED** [review]
IWOMM

File: include/Core/Context.hpp [hunk 2]
```diff
@@ -29,6 +29,7 @@ class Context {
     void SetExit(bool exit) { m_Exit = exit; }
     void SetWindowWidth(unsigned int width) { m_WindowWidth = width; }
     void SetWindowHeight(unsigned int height) { m_WindowHeight = height; }
+    void SetWindowIcon(const std::string path);
```
Comment (Dragonfly911117) [inline include/Core/Context.hpp:32]: Use const string ref saves resources.

File: src/Core/Context.cpp [hunk 3]
```diff
@@ -134,4 +134,9 @@ std::shared_ptr<Context> Context::GetInstance() {
     }
     return s_Instance;
 }
+
+void Context::SetWindowIcon(const std::string path) {
```
Comment (Dragonfly911117) [inline src/Core/Context.cpp:138]: Use const string ref saves resources.

**Review by Dragonfly911117 — APPROVED** [review]
LGTM&IWOMM

**Review by NOOBDY — APPROVED** [review]

**Pull request merged**